lib/elm-test.js

```diff
@@ -241,12 +245,12 @@ function main() {
     // commander would interpret the `tests` in `elm-test tests src` as a
     // command and only run tests in `src/`, ignoring all files in `tests/`.
     .command('__elmTestCommand__ [globs...]', { hidden: true, isDefault: true })
-    .action((testFileGlobs) => {
+    .action(async (testFileGlobs) => {
       const options = program.opts();
       const pathToElmBinary = getPathToElmBinary(options.compiler);
       const projectRootDir = getProjectRootDir('tests');
       const processes = Math.max(1, os.cpus().length);
-      RunTests.runTests(
+      await RunTests.runTests(
```

Comment:
Why was this change made? `.action()` does not take a function that returns a promise (`async` functions always return promises). Instead we run the promise and handle the results with `.then` and `.catch`. In general, one needs `.catch()` at the top-most level of your program, where your promise chain ends. Otherwise if the promise fails you’ll end up with an uncaught promise rejection.

Reply:
`.action()` does seem to take a function that returns a promise.

https://github.com/tj/commander.js/blob/6c26370fcba737d9786a34d0e8693f689a067569/lib/command.js#L1237
:arrow_down:
https://github.com/tj/commander.js/blob/6c26370fcba737d9786a34d0e8693f689a067569/lib/command.js#L1146-L1154

Reply:
Hmm, yes, that seems to be the case! I found this:

https://github.com/tj/commander.js#action-handler

> You may supply an `async` action handler, in which case you call `.parseAsync` rather than `.parse`.

But we use `.parse`. I wonder if I had a reason for that when adding `commander` 1.5 years ago, or if I simply did not know about `.parseAsync`.

Reply:
Should I change to `parseAsync` then?

Reply:
Maybe! I would have to play around with it locally to learn how it works.

Reply:
I played around a bit locally.

- You can remove `async` and `await` here – it’s not needed.
- I think we could simplify some code by using `async` action handlers and `parseAsync`, but that’s out of scope for this PR.

Reply:
> You can remove async and await here – it’s not needed.

Any reasoning for this? I'm using async/await in the `Install.installUnstableTestMaster` function. Why is it not needed in its usages?

Reply:
Ah, I've missed the `.then(exit(0),exit(1))` afterwards. Makes sense!